src/eh1/serial.rs

```diff
@@ -0,0 +1,636 @@
+//! Serial mock implementations.
+//!
+//! You can set expectations for serial read and write transactions on a mock
+//! Serial device. Creating error transactions is supported as well.
+//!
+//! Note that the `embedded_hal` crate provides both non-blocking and blocking
+//! serial traits. You can use the same mock for both interfaces.
+//!
+//! ## Usage: Non-blocking serial traits
+//!
+//! ```
+//! # use eh1 as embedded_hal;
+//! // Note that we're using the non-blocking serial traits
+//! use embedded_hal_nb::serial::{Read, Write};
+//! use embedded_hal_mock::eh1::serial::{
+//!     Mock as SerialMock,
```

Comment:
this might be how we've always done it but, it seems like it's worth having descriptive names at the module level to avoid every consumer having to `Mock as SerialMock` ?

Reply:
I'd rather keep PRs focussed on one thing, makes it easier to review 🙂

The original idea was to implement the same pattern for every mock. However, I guess you're right and if every user aliases the import, we can rename the mocks as well. So: A separate PR would be welcome (but it might make sense to base it on this branch, or to wait until it is merged.)